data/json/recipes/weapon/mods.json

```diff
@@ -65,5 +65,96 @@
       [ [ "scrap", 1 ] ],
       [ [ "plastic_chunk", 1 ] ]
     ]
+  },
+  {
+    "type": "recipe",
+    "result": "red_dot_sight",
```

Comment:
From wikipedia https://en.wikipedia.org/wiki/Red_dot_sight
```The typical configuration for a red dot sight is a tilted spherical mirror reflector with a red light-emitting diode (LED) at its off axis focus. The mirror has a partially silvered multilayer dielectric dichroic coating designed to reflect just the red spectrum allowing it to pass through most other light.```
In short, the actual device requires a *very specific* lens, but you've substituted an arbitrary lens.
In other words, if you take a lens off of a pair of glasses, pair of binoculars, magnifying glass, or literally any other source of the lens item in the game, it will not be the right kind of lens to make a red dot sight.